### PR TITLE
6348 - Deunify FilesInUse messages in Burn

### PR DESCRIPTION
**Pull request by rseanhall**
Fixes https://github.com/wixtoolset/issues/issues/6348

### CODE REVIEW

File: src/api/burn/balutil/inc/IBootstrapperApplication.h [hunk 1]
```diff
@@ -488,27 +488,19 @@ DECLARE_INTERFACE_IID_(IBootstrapperApplication, IUnknown, "53C31D56-49C0-426B-A
         __inout int* pResult
         ) = 0;
 
-    // OnExecuteFilesInUse - called when the engine encounters files in use while
-    //                       executing a package.
+    // OnExecuteFilesInUse - called when the engine receives a files in use message
+    //                       while executing a package.
     //
-    // Return:
```
Comment (barnson) [inline src/api/burn/balutil/inc/IBootstrapperApplication.h:494]: Why remove these? This is the only "doc" without spelunking source.

File: src/api/burn/WixToolset.Mba.Core/IBootstrapperApplication.cs [hunk 2]
```diff
@@ -1450,6 +1445,25 @@ public enum ErrorType
...
+    public enum FilesInUseType
+    {
+        /// <summary>
+        /// Generated from INSTALLMESSAGE_FILESINUSE.
+        /// </summary>
+        Msi,
+        /// <summary>
+        /// Generated from INSTALLMESSAGE_RMFILESINUSE.
+        /// </summary>
+        MsiRm,
```
Comment (barnson) [inline src/api/burn/WixToolset.Mba.Core/IBootstrapperApplication.cs:1460]: Should we be spelling out Restart Manager (here and/or in the native code)?

File: src/api/burn/balutil/inc/IBootstrapperApplication.h (hunk 1 above)
Comment (rseanhall) [reply src/api/burn/balutil/inc/IBootstrapperApplication.h:494]: You mean source code? Burn doesn't own how the return codes are processed. The BA needs to look at the source of the FilesInUse message and return a value based on that source's documentation which is provided as a link here.

File: src/api/burn/WixToolset.Mba.Core/IBootstrapperApplication.cs (hunk 2 above)
Comment (rseanhall) [reply src/api/burn/WixToolset.Mba.Core/IBootstrapperApplication.cs:1460]: I guess I was following Microsoft's lead here, they always used RM - `INSTALLMESSAGE_RMFILESINUSE`, `MsiRMFilesInUse` (Dialog), etc.

File: src/api/burn/balutil/inc/IBootstrapperApplication.h (hunk 1 above)
Comment (barnson) [reply src/api/burn/balutil/inc/IBootstrapperApplication.h:494]: I didn't realize it was just pass-through. Hopefully we'll never need more than an int...

File: src/api/burn/WixToolset.Mba.Core/IBootstrapperApplication.cs (hunk 2 above)
Comment (barnson) [reply src/api/burn/WixToolset.Mba.Core/IBootstrapperApplication.cs:1460]: Yeah, that's a typical Microsoft native-code approach. Not so much with managed code, though. And Burn hasn't shied away from long-ass names...

File: src/api/burn/balutil/inc/IBootstrapperApplication.h (hunk 1 above)
Comment (rseanhall) [reply src/api/burn/balutil/inc/IBootstrapperApplication.h:494]: I guess it's not exactly pass-through, there's still some coercion by calling `UserExperienceCheckExecuteResult`. I wasn't sure whether to remove that, too.

Comment (barnson) [reply src/api/burn/balutil/inc/IBootstrapperApplication.h:494]: Seems that it should be completely pass-through (though we still want to be able to talk to Burn) or translate Burn return values to the underlying engine (as Burn is hard-coded for the ones it supports).

**Pull request merged**